addon/-private/extensions/with-validation.js

```diff
@@ -21,10 +24,17 @@ export function withExtension(klass) {
     init(...args) {
       super.init(...args);
 
-      const validations = getValidationsFor(this.constructor);
+      const { constructor } = this;
+
+      const validations = getValidationsFor(constructor);
+      const meta = Ember.meta(this);
 
       for (let key in validations) {
-        wrapField(this.constructor, this, validations, key);
+        const validation = validations[key];
+
+        wrapComputedProperty(constructor, this, meta, validation, key);
+
+        runValidation(validation, constructor, key, this[key], 'init');
```

Comment:
I'm not sure we need to run validations here, since class fields and getters/setters will trigger when the values are set anyways. I suppose the `init` part of the error message is still somewhat valuable, but it probably won't show up anyways since the error for the accessor will trigger first.

Reply:
That would remove the `init` validations altogether, right?

I'm fine with that, personally -- I think it probably makes more sense just to run the validations on `get`/`set`

Reply:
So, I looked into this and I'm not sure that it'll make sense to remove the `init` validation.

I tried removing the validator on `init` and adding a validator to `get`. This had the side-effect of making it impossible to check that the props passed to the object match the expected validation, because as part of initializing the object in `.create()`, the initial value is accessed. Basically, the validation on `get` will be called before the property is overwritten with the new value, which doesn't work.